test/IntegrationTests/SmokeTests.cs

```diff
@@ -271,8 +273,84 @@ public void LogsNoneInstrumentations()
 
         collector.AssertEmpty(5.Seconds());
     }
+
 #endif
 
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void TracesNoneInstrumentations()
+    {
+        using var collector = new MockSpansCollector(Output);
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_TRACES_ENABLED_INSTRUMENTATIONS", "none");
+        RunTestApplication();
+        collector.AssertEmpty(1.Seconds());
+    }
+
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void MetricsNoneInstrumentations()
+    {
+        using var collector = new MockMetricsCollector(Output);
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS", "none");
+        RunTestApplication();
+        collector.AssertEpmty(1.Seconds());
+    }
+
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void LogsDisabledInstrumentation()
+    {
+        using var collector = new MockLogsCollector(Output);
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_LOGS_DISABLED_INSTRUMENTATIONS", "ILogger");
+        EnableBytecodeInstrumentation();
+        RunTestApplication();
+        collector.AssertEmpty(1.Seconds());
+    }
+
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void MetricsDisabledInstrumentation()
+    {
+        using var collector = new MockMetricsCollector(Output);
+        SetExporter(collector);
+
+        StringBuilder sbToDisableMetrics = new StringBuilder();
+        sbToDisableMetrics.Append(MetricInstrumentation.NetRuntime.ToString())
+        .Append(",").Append(MetricInstrumentation.HttpClient.ToString());
+
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_METRICS_DISABLED_INSTRUMENTATIONS", sbToDisableMetrics.ToString());
+        EnableBytecodeInstrumentation();
+        RunTestApplication();
+        Assert.False(
+            collector.CheckInstrumentationScopeIsAbsent(
+            new List<string>() { "OpenTelemetry.Instrumentation.Process" },
+            1.Seconds()),
+            "Didn't find metrics instrumentation while it should be present.");
+        Assert.True(
+            collector.CheckInstrumentationScopeIsAbsent(
+            new List<string>() { "OpenTelemetry.Instrumentation.NetRuntime", "OpenTelemetry.Instrumentation.HttpClient" },
+            1.Seconds()),
+            "Found Metrics instrumentation while it should be disabled.");
```

Comment:
This can lead to flakiness.

Each `CheckInstrumentationScopeIsAbsent` is using `_metricsSnapshots.GetConsumingEnumerable()`. Take notice that whatever the first call of `CheckInstrumentationScopeIsAbsent` will process then it won't be processed by the second call of `CheckInstrumentationScopeIsAbsent`. Because of that we can have false positives (passing tests which should fail) if there is a bug in production code.

Personally, I think it would be good enough to have a test like this:

```csharp
    [Fact]
    [Trait("Category", "EndToEnd")]
    public void MetricsDisabledInstrumentation()
    {
        using var collector = new MockMetricsCollector(Output);
        SetEnvironmentVariable("OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS", "HttpClient");
        SetEnvironmentVariable("OTEL_DOTNET_AUTO_METRICS_DISABLED_INSTRUMENTATIONS", "HttpClient");
        EnableBytecodeInstrumentation();
        RunTestApplication();
        collector.AssertEmpty(1.Seconds());
    }
```

Other tests are checking if `OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS` is working. This tests would test if `OTEL_DOTNET_AUTO_METRICS_DISABLED_INSTRUMENTATIONS` is working and also if it has higher priority than `OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS`. We would also not need to implement something like `CheckInstrumentationScopeIsAbsent` 😉

Reply:
thanks ! I would say this is because my lack of knowledge... I thought all Metrics Instrumentation are enabled by default and I need to set and check that's not present... didn't know if you set "OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS", "HttpClient" , only that metrics would be selected rather than all.